sufia-models/lib/sufia/models/utils.rb

```diff
@@ -1,12 +1,26 @@
+# This module includes a utility methods
 module Sufia
   module Utils
     extend ActiveSupport::Concern
 
+    # retry the block if the conditional call is true unless we hit the maximum tries
+    #
+    # @param number_of_tries [Fixnum] maximum number of times to retry the block
+    # @param condition [Boolean] conditional to call and see if we SHOULD retry
+    #
+    # @return result of the block call
     def retry_unless(number_of_tries, condition, &block)
       self.class.retry_unless(number_of_tries, condition, &block)
     end
 
     module ClassMethods
+
+      # retry the block if the conditional call is true unless we hit the maximum tries
+      #
+      # @param number_of_tries [Fixnum] maximum number of times to retry the block
+      # @param condition [Boolean] conditional to call and see if we SHOULD retry
+      #
```

Comment:
Would it be helpful to write anything about the block using something like https://gist.github.com/chetan/1827484#blocks?